dev/create_release_tag.py

```diff
@@ -39,17 +39,10 @@
     "--dry-run/--no-dry-run", is_flag=True, default=True, show_default=True, envvar="DRY_RUN"
 )
 def main(new_version: str, remote: str, dry_run: bool = False):
-    if not dry_run:
-        version = Version(new_version)
-        release_branch = f"branch-{version.major}.{version.minor}"
-        current_branch = subprocess.check_output(
-            ["git", "branch", "--show-current"], text=True
-        ).strip()
-        assert (
-            current_branch == release_branch
-        ), f"Expected to be on {release_branch} but on {current_branch}"
```

Comment:
In our internal job, this script is executed on a random branch cut from a release branch and the assertion here doesn't pass.